config/initializers/warden.rb

```diff
@@ -0,0 +1,17 @@
+
+Rails.application.config.middleware.use Warden::Manager do |manager|
+  manager.failure_app = proc { |_env| ['401', { 'Content-Type' => 'application/json' }, { error: 'Unauthorized', code: 401 }] }
+end
+
+Warden::Manager.serialize_into_session do |user|
+  # convert to warden_session_obj
+  user.slice("id", "role")
+end
+
+Warden::Manager.serialize_from_session do |warden_session_obj|
+  User.find_by(id: warden_session_obj["id"],
+               role: warden_session_obj["role"])
+end
+
+# TODO: DAVID - Put user login counters here
```

Comment:
nit: could you link to the JIRA and date this TODO?

Reply:
Just realized that I left all these TODOs behind by mistake. I'm reviewing them all.